bigtable/client/table.cc

```diff
@@ -152,19 +161,84 @@ RowReader Table::ReadRows(RowSet row_set, std::int64_t rows_limit,
                        bigtable::internal::ReadRowsParserFactory>());
 }
 
-std::pair<bool, Row> Table::ReadRow(std::string row_key, Filter filter) {
+std::pair<bool, Row> Table::ReadRow(std::string row_key, Filter filter,
+                                    grpc::Status& ret_status) {
   RowSet row_set(std::move(row_key));
   std::int64_t const rows_limit = 1;
   RowReader reader =
-      ReadRows(std::move(row_set), rows_limit, std::move(filter));
+      ReadRows(std::move(row_set), rows_limit, std::move(filter), ret_status);
+  if (not ret_status.ok()) {
+    return std::make_pair(false, Row("", {}));
+  }
   auto it = reader.begin();
   if (it == reader.end()) {
+    grpc::Status status = reader.finish();
+    if (not status.ok()) {
+      ret_status = grpc::Status(status.error_code(), status.error_message());
```

Comment:
Or just `ret_status = status`?  Or `ret_status = reader.finish(); if (not ret_status.ok()) { return ..; }`